streamz/tests/test_core.py

```diff
@@ -774,33 +775,82 @@ def test_destroy():
 def test_stream_name():
     ''' Test that stream_name is passed correctly to the core methods.'''
     test_name = "some test name"
+
     sin = Stream(stream_name=test_name)
     sin2 = Stream()
 
+    assert sin.name == test_name
+    # when not defined, should be None
+    assert sin2.name is None
+
     # add new core methods here, initialized
+    # these should be functions, use partial to partially initialize them
+    # (if they require more arguments)
     streams = [
-               sin.map(lambda x : x, stream_name=test_name),
-               sin.accumulate(lambda x1, x2 : x1, stream_name=test_name),
-               sin.filter(lambda x : True, stream_name=test_name),
-               sin.partition(2, stream_name=test_name),
-               sin.sliding_window(2, stream_name=test_name),
-               sin.timed_window(.01, stream_name=test_name),
-               sin.rate_limit(.01, stream_name=test_name),
-               sin.delay(.02, stream_name=test_name),
-               sin.buffer(2, stream_name=test_name),
-               sin.zip(sin2, stream_name=test_name),
-               sin.combine_latest(sin2, stream_name=test_name),
-               sin.frequencies(stream_name=test_name),
-               sin.flatten(stream_name=test_name),
-               sin.unique(stream_name=test_name),
-               sin.union(stream_name=test_name),
-               sin.pluck(0, stream_name=test_name),
-               sin.collect(stream_name=test_name),
+               partial(sin.map, lambda x : x),
```

Comment:
this is not completely necessary. my hopes of using partial was so i could reuse the list for the `bad_kwargs` test. however, I found I need to omit `map` and `scan` methods (or methods that call it) anyway.